File: LeetCode/cpp/1023.cpp
class Solution {
public:
    vector<bool> camelMatch(vector<string> &queries, string &pattern) {
        vector<bool> ret;
        int n = pattern.size();
        for (auto &query:queries) {
            bool res = true;
            int i = 0;
            for (auto &q:query) {
                if (q < 'Z') {
                    if (i >= n || q != pattern[i]) {
                        res = false;
                        break;
                    }
                    ++i;
                } else if (q == pattern[i])
                    ++i;
            }
            ret.push_back(res && i == n);
        }
        return ret;
    }
};

File: LeetCode/python3/1023.py
class Solution:
    def camelMatch(self, queries: List[str], pattern: str) -> List[bool]:
        n, ret = len(pattern), []
        for query in queries:
            i, res = 0, True
            for q in query:
                if q <= 'Z':
                    if i >= n or q != pattern[i]:
                        res = False
                        break
                    i += 1
                elif i < n and q == pattern[i]:
                    i += 1
            ret.append(res and i == n)
        return ret
